google-ads-migration-examples/src/main/java/com/google/ads/googleads/migration/createcampaign/CreateCompleteCampaignBothApisPhase2.java

```diff
@@ -0,0 +1,557 @@
+
+    // Create the operation.
+    CampaignOperation op = CampaignOperation.newBuilder().setCreate(campaign).build();
+
+    // Get the Campaign service.
+    try (CampaignServiceClient campaignServiceClient =
+        googleAdsClient.getLatestVersion().createCampaignServiceClient()) {
+      // Add the campaign.
+      MutateCampaignsResponse response =
+          campaignServiceClient.mutateCampaigns(Long.toString(customerId), ImmutableList.of(op));
+      String campaignResourceName = response.getResults(0).getResourceName();
+      // Retrieve the campaign.
+      Campaign newCampaign = getCampaign(googleAdsClient, customerId, campaignResourceName);
+      // Display the results.
+      System.out.printf(
+          "Campaign with ID %s and name '%s' was created.%n",
+          newCampaign.getId().getValue(), newCampaign.getName().getValue());
+      return newCampaign;
+    }
+  }
+
+  /**
+   * Retrieves the campaign budget.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param budgetResourceName resource name of the new campaign budget.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private CampaignBudget getBudget(
+      GoogleAdsClient googleAdsClient, long customerId, String budgetResourceName) {
+    // Get the GoogleAdsService.
+    try (GoogleAdsServiceClient googleAdsServiceClient =
+        googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+
+      // Create the request.
+      SearchGoogleAdsRequest request =
+          SearchGoogleAdsRequest.newBuilder()
+              .setCustomerId(Long.toString(customerId))
+              .setPageSize(PAGE_SIZE)
+              .setQuery(
+                  String.format(
+                      "SELECT campaign_budget.id, campaign_budget.name, "
+                          + "campaign_budget.resource_name FROM campaign_budget "
+                          + "WHERE campaign_budget.resource_name = '%s'",
+                      budgetResourceName))
+              .build();
+
+      // Retrieve the budget.
+      SearchPagedResponse searchPagedResponse = googleAdsServiceClient.search(request);
+      return searchPagedResponse.getPage().getResponse().getResults(0).getCampaignBudget();
+    }
+  }
+
+  /**
+   * Creates a budget.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private CampaignBudget createBudget(GoogleAdsClient googleAdsClient, long customerId) {
+    // Create the budget.
+    CampaignBudget budget =
+        CampaignBudget.newBuilder()
+            .setName(StringValue.of("Interplanetary Cruise Budget #" + System.currentTimeMillis()))
+            .setDeliveryMethod(BudgetDeliveryMethod.STANDARD)
+            .setAmountMicros(Int64Value.of(500_000))
+            .build();
```

Comment:
It's a bit weird that the CPC bid is 10MM and the budget is 500K. Not a big deal but this doesn't really make any kind of sense.

Reply:
Agreed, nice catch. I've flipped them.